modules/compute/vm-instance/main.tf

```diff
@@ -230,5 +230,16 @@ resource "google_compute_instance" "compute_vm" {
     ignore_changes = [
       metadata["ssh-keys"],
     ]
+
+    precondition {
+      condition     = (length(var.network_interfaces) == 0) != (var.network_self_link == null && var.subnetwork_self_link == null)
+      error_message = "Exactly one of network_interfaces or network_self_link/subnetwork_self_link must be specified."
+    }
+
+    precondition {
```

Comment:
I do not believe that the 2nd precondition should be included. It breaks the following pattern:

```yaml
deployment_groups:
- group: first
  modules:
  - id: network1
    source: modules/network/pre-existing-vpc
  - id: simplevm
    source: modules/compute/vm-instance
    use:
    - network1
    settings:
      machine_type: e2-medium
      add_deployment_name_before_prefix: true
    outputs:
    - internal_ip
```

I don't think we should worry about network_self_link/subnetwork_self_link "disagreeing" if they are both supplied.